src/lines-component.coffee

```diff
@@ -19,11 +20,7 @@ class LinesComponent
   placeholderTextDiv: null
 
   constructor: ({@presenter, @hostElement, @useShadowDOM, visible}) ->
```

Comment:
We may still introduce a couple of improvements here:

1. Avoid allocating a `TileComponent` object whenever new tiles need to be rendered. This will remove some pressure from GC and will improve performance as well. I somewhat liked the idea of having a tile encapsulated in its own component, but this area of the code is particularly performance sensitive and we should pay attention to each allocation.
2. When more than one tile gets rendered on screen for a given frame, we observe multiple `Parse HTML` events (one for each new tile). An alternative approach could be to parse all the lines at the same time and append them later to their respective tile. Not sure if the performance implications here are very serious, but this could make us save some cycles.
3. Avoid removing tile nodes while scrolling. Removing a tile node from the DOM takes between 0.5 and 1.0 ms: for performance sensitive scenarios like scrolling, we may simply put `display: none` and cleanup dangling tiles once scrolling ends.

Not sure if we should tackle 3), but I feel pretty confident we should have a look at 1) and 2)